docs/reference/how-to/size-your-shards.asciidoc

```diff
@@ -314,3 +313,53 @@ POST _reindex
   }
 }
 ----
+
+[discrete]
+[[troubleshoot-shard-related-errors]]
+=== Troubleshoot shard-related errors
+
+Here’s how to resolve common shard-related errors.
+
+[discrete]
+==== this action would add [x] total shards, but this cluster currently has [y]/[z] maximum shards open;
+
+The <<cluster-max-shards-per-node,`cluster.max_shards_per_node`>> cluster
+setting limits the maximum number of open shards for a cluster. This error
+indicates an action would exceed this limit.
+
+If you're confident your changes won't destabilize the cluster, you can
+temporarily increase the limit using the <<cluster-update-settings,update
+cluster settings API>> and retry the action.
+
+[source,console]
+----
+PUT _cluster/settings
+{
+  "transient" : {
+    "cluster.max_shards_per_node": 1200
+  }
+}
+----
+
+This increase should only be temporary. As a long-term solution, we recommend
+you add non-frozen data notes or <<reduce-cluster-shard-count,reduce your
```

Comment:
I wonder if rather than `non-frozen` we should dictate to add nodes for the tier that is oversharded? While today, it is correct that adding a cold node will avoid the limit also for all your hot data, it is not really a solution to the underlying oversharding problem of having too many hot shards for the number of hot nodes.

Reply:
Good point! I'll revise this to:

> we recommend you add nodes to the oversharded data tier or ...